src/ast/passes/semantic_analyser.cpp

```diff
@@ -2969,45 +2976,53 @@ void SemanticAnalyser::visit(VarDeclStatement &decl)
         << "Invalid variable declaration type: " << decl.var->type;
   }
 
-  // This should be the first time we're seeing this variable
-  if (auto decl_search = variable_decls_[scope_].find(var_ident);
-      decl_search != variable_decls_[scope_].end()) {
-    if (decl_search->second != decl.loc) {
-      // Note this will change if/when proper block/lexical scoping is added
-      LOG(ERROR, decl.loc, err_)
-          << "Variable " << var_ident << " was already declared.";
-      LOG(ERROR, decl_search->second, err_) << "Initial declaration";
+  if (is_first_pass()) {
```

Comment:
How come this condition was not necessary before?

Reply:
Since variables are no longer probe scoped. This is now valid `BEGIN { if (1) { let $x; } else { let $x; } let $x; }` - See how we define the last $x in the outer scope at the end. Now on subsequent passes, it's viewed as variable shadowing in the if/else blocks.

Reply:
Right, makes sense. A comment would be helpful, though.